macros/shopify_partition_by_cols.sql

```diff
@@ -0,0 +1,15 @@
+{% macro shopify_partition_by_cols(base_col, source_relation='source_relation') %}
+
+{{ adapter.dispatch('shopify_partition_by_cols', 'shopify') (base_col, source_relation) }}
+
+{%- endmacro %}
+
+{% macro default__shopify_partition_by_cols(base_col, source_relation='source_relation') %}
+
+    {%- if var('shopify_union_schemas', false) or var('shopify_union_databases', false) -%}
+    {{ base_col }}, {{ source_relation }}
+    {%- else %}
+    {{ base_col }}
+    {%- endif %}
+
+{% endmacro %}
```

Comment:
This is 100% for redshift since it was giving the constant expressions error after I switched the source tables to views. I chose this way since it is a repetitive need, but also allows the code in the model to be readable. The other option is to go back to making all the source models as tables.